File: C++/A/248A-cupboards.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, r, l;
    cin >> n;
    vector<int> right, left;
    while (n--)
    {
        cin >> r >> l;
        right.push_back(r);
        left.push_back(l);
    }
    int rightopen = count(right.begin(), right.end(), 1);
    int rightclosed = count(right.begin(), right.end(), 0);
    int leftopen = count(left.begin(), left.end(), 1);
    int leftclosed = count(left.begin(), left.end(), 0);
    cout << min(rightopen, rightclosed) + min(leftopen, leftclosed);
}

File: Python/248A-cupboards.py
n = int(input())
right = []
left = []
for i in range(n):
    r, l = map(int, input().split())
    right.append(r)
    left.append(l)
righttime = min(right.count(0), right.count(1))
lefttime = min(left.count(0), left.count(1))
print(righttime + lefttime)
